linode/manager
# refactor: [M3-8712] - Tanstack Reat Router rollout setup + betas routes migration

## Description 📝
This PR offers a safe pattern to be able to slowly roll out the new Tanstack routing migration.
The worry was to have to replace all of the routes and routing internals at once, resulting in a very large, hard to manage feature branch. Fortunately, this is not the case! The way we can incrementally implement the new routing is to accept to have the two routers cohabitating, and implement a cascading routing approach:

- `react-router-dom`:
  - serve non-migrated routes
  - route not found: defer to `migrationRouter` (Tanstack)
- `tanstack-react-router`
  - server migrated routes
  - route not found: render `<NotFound />`

The deferring logic lives in `<MainContent />` where a wildcard catch-all will defer all unhandled routes to the tanstack `<RouterProvider />`. The POC is showing that there's no conflict resulting in this approach. 

The `<Link />` components will be replaced last to avoid any potential conflict and keep things simple (already confirmed that the current <Link to="" /> behavior is not causing any problem navigating to the new routes 🎆 

Lastly, the PR introduces development tools to make out task easier and safer (linting, testing, inspecting)

## Changes  🔄
- Implement router deferring logic
- Migrate /betas paths to the new routing
  - remove react-router-dom handling of Betas paths
  - replace utils
  - switch from using state navigation to params navigation for `betas/<detail>`
    - improve betas/{id} query to be able to be disabled
  - update tests
- Add a new e2e smoke test for beta
- Update mockServiceWorker declaration
- Enable routes migration test
- Improve path utils
- Add missing lazy imports in a few files to remove build error messages

## How to test 🧪

### Verification steps
- Pull code locally and update dependencies
- Confirm the /betas feature has no regression
- Confirm application routing behaves well everywhere else
- Confirm tests are passing
- Confirm being able to use the routing debugging tools in our DevTools

## As an Author I have considered 🤔

*Check all that apply*
